contracts/ColonyNetwork.sol

```diff
@@ -245,30 +237,9 @@ contract ColonyNetwork is ColonyNetworkStorage {
   calledByColony
   skillExists(_skillId)
   {
-    uint reputationUpdateLogLength = reputationUpdateLogs[activeReputationUpdateLog].length;
-    uint nPreviousUpdates = 0;
-    if (reputationUpdateLogLength > 0) {
-      nPreviousUpdates = reputationUpdateLogs[activeReputationUpdateLog][reputationUpdateLogLength-1].nPreviousUpdates + reputationUpdateLogs[activeReputationUpdateLog][reputationUpdateLogLength-1].nUpdates;
-    }
-    uint nUpdates = (skills[_skillId].nParents + 1) * 2;
-    if (_amount < 0) {
-      //TODO: Never true currently. _amount needs to be an int.
-      nUpdates += 2 * skills[_skillId].nChildren;
-    }
-    reputationUpdateLogs[activeReputationUpdateLog].push(ReputationLogEntry(
-      _user,
-      _amount,
-      _skillId,
-      msg.sender,
-      nUpdates,
-      nPreviousUpdates));
-  }
-
-  function getReputationUpdateLogLength(bool active) public view returns (uint) {
-    uint logIdx = activeReputationUpdateLog;
-    if (!active) {
-      logIdx = (logIdx + 1 ) % 2;
-    }
-    return reputationUpdateLogs[logIdx].length;
+    uint nParents = skills[_skillId].nParents;
+    // TODO: Is it cheaper to check if _amount is <0, and if not, just set nChildren to 0, because children won't be updated for such an update?
+    uint nChildren = skills[_skillId].nChildren;
+    ReputationMiningCycle(nextReputationMiningCycle).appendReputationUpdateLog(_user, _amount, _skillId, msg.sender, nParents, nChildren);
```

Comment:
From `Colony`? There would still be a call to `ColonyNetwork` to make sure that the calling contract was a colony every time, so I don't think you gain much. If you want to save gas here, I think the approach to take is to pass arrays to `appendReputationUpdateLog` that are then iterated (but good iteration!) over in `ReputationMiningCycle`, so you only have one call for a block of updates. You would go from eight external calls to two, so that should save at least 120000 gas, right?

Reply:
Do we have a limit on the number of domains and skills on a task?

Reply:
Currently only one of each. I don't see multiple domains happening, but I can see a desire for multiple skills in the future. But that introduces problems regardless of which implementation discussed here, as far as I can see?

Reply:
I can't find a call for a limit on number of skills in the paper, maybe add it to the list of missing bits in that WP repo issue. And yes, it's a problem either way but it doesn't have to be solved here for the MVP.